scripts/spack/packages/axom/package.py

```diff
@@ -104,21 +150,35 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     variant(
         "profiling",
         default=False,
-        when="@develop",
-        description="Build with hooks for Adiak/Caliper performance analysis",
+        when="@:0.12.0",
+        description="Build with hooks for Adiak/Caliper performance analysis. " \
+                    "Deprecated -- use the adiak and/or caliper variants directly.",
     )
```

Comment:
The `profiling` variant added a dependency on `adiak` and `caliper`. Since `sina` has a soft dependency on `adiak`, I replaced it with direct variants for `adiak` and `caliper`.

Should we remove the `profiling` variant? In this PR I added a description that this variant in deprecated and changed the `when` line.  

Is there a better way to do this?  I noticed this open issue on spack https://github.com/spack/spack/issues/38829